extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaConsumerConfigs.java

```diff
@@ -38,7 +38,6 @@
     props.put("group.id", StringUtils.format("kafka-supervisor-%s", IdUtils.getRandomId()));
     props.put("auto.offset.reset", "none");
     props.put("enable.auto.commit", "false");
-    props.put("isolation.level", "read_committed");
```

Comment:
I think to remove `isolation.level` here and leaves the configuration to users changes the current behavior. 

Currently no user cares nothing about this configuration when using a higher version of kafka such as 0.11. By removing it, they need to set this property in supervisor spec,  or the default value, which is `read_uncommitted`, will be applied, which may be not what they expect.

If this is the root cause that limits the Druid to use Kafka lower than 0.11, I think maybe we can introduce another property, as the same way as `pollTimeout` property does, then we can unset `isolation.level` property according to the value of this new property. Only those who want to use Druid with older kafka  need to set this new property.

Reply:
Yes, change the current behavior is what I want. 

You are right, most people really don't care this Kafka configuration(Let it be default behavior). As I know the default logic of the Kafka Producer is not to enable transactions feature and the same as Kafka Consumer. So that maybe Druid Kafka indexing service keep the same default logic is more reasonable.

If a Druid user want Druid to consume transactional Kafka topics, I think it is more reasonable to let users set this parameter in consumerProperties like 'bootstrap.servers' because he knows what he is going to do and why.

By the way, Druid from 0.15 to better can't consume old version Kafka is really confused me(I believe it's not just me). Generally speaking, higher Kafka consumer client is able to consume old version Kafka cluster unless it involves high-version-specific api.

Thanks for reviewing!

Reply:
We don't know whether or not the users have enabled transactional message in their clusters, so changing the current behavior may cause backward compatibility problem.

Reply:
Makes sense! I add a new property named 'consumeTransactionally' in consumerProperties, which is a Kafka consumer level property to control 'isolation.level';
If users don't set 'consumeTransactionally' or set 'consumeTransactionally' true, druid will consume Kafka Transactionally by default.
Set 'consumeTransactionally' false here can disable druid to consume Kafka Transactionally, which means druid can consume lower version of Kafka now like 0.10.x

Reply:
@FrankChen021  Hi Frank, sorry to bother you. I have tested this change in our Druid cluster recently. What should I do next ?